main.go

```diff
@@ -316,6 +320,80 @@ Note: may not work with some transports such as p2p-circuit (not applicable) and
 							for _, p := range resp.Protocols {
 								fmt.Printf("\t- %q\n", p)
 							}
+							return nil
+						},
+					}, {
+						Name:      "ping",
+						ArgsUsage: "<multiaddr>",
+						Flags: []cli.Flag{
+
+							&cli.BoolFlag{
+								Name:        "direct",
+								Usage:       `Ping the peer directly rather than through a relay.`,
+								DefaultText: "false",
+								Value:       false,
+							},
+						},
+						Usage:       "ping a peer",
+						Description: "connects to the target address and pings",
+						Action: func(c *cli.Context) error {
+							if c.NArg() != 1 {
+								return fmt.Errorf("invalid number of arguments")
+							}
+							maStr := c.Args().First()
+							ai, err := peer.AddrInfoFromString(maStr)
+							if err != nil {
+								return err
+							}
+							// Set the activation threshold to 1 so that we learn our own address with just one connection. This lets us holepunch sooner.
```

Comment:
This is "at all" rather than sooner right? Since there's no source of address discovery other than from the address of the one peer that's passed in.